torch/__init__.py

```diff
@@ -185,6 +185,36 @@ class ByteStorage(_C.ByteStorageBase, _StorageBase):
     pass
 
 
+# Numpy array interface, to support `numpy.asarray(tensor) -> ndarray`
+def __array__(self, dtype=None):
+    if dtype is None:
+        return self.cpu().numpy()
+    else:
+        return self.cpu().numpy().astype(dtype, copy=False)
+
+
+# Wrap Numpy array again in a suitable tensor when done, to support e.g.
+# `numpy.sin(tensor) -> tensor` or `numpy.greater(tensor, 0) -> ByteTensor`
+def __array_wrap__(self, array):
+    if array.ndim == 0:
```

Comment:
for later: we will be adding support for 0-dim tensors in PyTorch at some point, so maybe it would be good to add a TODO here to modify this once it's implemented?

Reply:
Sure, I'll add a TODO.